### PR TITLE
new bitset

### PR DESCRIPTION
**Pull request by Rhazarian**
Bitset is now superfast - only one native call (and nothing more) per operation.
Powers of 2 are now precomputed and new bitwise natives are used instead of some mathematical magic.
Added `flip` method, which allows to change the state of a bit to the contrary one.
Added new tests.

### CODE REVIEW

**Comment by Frotty** [comment]
LGTM

File: wurst/data/BitSet.wurst [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
-	return not this.containsPow(pow) ? bitset(this.val + pow) : this
+	return bitset(BlzBitOr(this.val, pows[v]))
 
 /** removes value v from the bitset */
 public function bitset.remove(int v) returns bitset
-	let pow = 2 .pow(v)
-	return this.containsPow(pow) ? bitset(this.val - pow) : this
+	return bitset(BlzBitAnd(this.val, reversePows[v]))
+
+/** adds value v to the bitset if the bitset does not contain it, removes it otherwise */
```
Comment (Cokemonkey11) [inline wurst/data/BitSet.wurst]: What domain of v is acceptable?

File: wurst/data/BitSet.wurst [hunk 2]
```diff
@@ -47,3 +62,28 @@ public function bitset.remove(int v) returns bitset
 
 @test function testRemove()
 	bitset(45).remove(3).val.assertEquals(37)
+
+@test function testFlip()
+	emptyBitset().add(3).flip(3).contains(3).assertFalse()
+	emptyBitset().flip(3).contains(3).assertTrue()
```
Comment (Cokemonkey11) [inline wurst/data/BitSet.wurst]: Maybe add some tests for boundary conditions

Comment (Rhazarian) [reply wurst/data/BitSet.wurst]: I've added a test.

File: wurst/data/BitSet.wurst (hunk 1 above)
Comment (Rhazarian) [reply wurst/data/BitSet.wurst]: It is now stated by the constant above.

**Comment by Rhazarian** [comment]
I have redesigned the API to make it more modern. It's now more similar to Java & C++ bitsets.
Old method names are kept for compatibility.

**Comment by Frotty** [comment]
Cool, is it done yet?

> Old method names are kept for compatibility.

If they duplicate functionality, you should deprecate them.

**Comment by Rhazarian** [comment]
Yea, I guess it's done.

**Comment by Frotty** [comment]
Yo, by deprecating old functions I means adding `@Deprecated("use func x instead").
NOT deleting them...
That just breaks everything.

**Comment by Rhazarian** [comment]
I did exactly that?

**Comment by Frotty** [comment]
Ah okay, I searched for `contains(int v)` but there was only a deletion match since you renamed the parameter..
Alright, thanks 👍
